### PR TITLE
Fix don't fail if directory already exists

### PR DESCRIPTION
**Pull request by drorasaf**
directory can already exist if there are dependency of multiple executables on setuptools or any other data files

### CODE REVIEW

File: PyInstaller/building/toc_conversion.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def __collect_data_files_from_zip(zipfilename):
         # 'PyInstaller.config' cannot be imported as other top-level modules.
         from ..config import CONF
         workpath = os.path.join(CONF['workpath'], os.path.basename(zipfilename))
-        os.makedirs(workpath)
+        os.makedirs(workpath, exist_ok=True)
```
Comment (htgoebel) [inline PyInstaller/building/toc_conversion.py]: I'm sorry, but this is Python-3-only :-(
So wrapping it in try/except, catching the IOError would be a more portable solution.

**Comment by drorasaf** [comment]
I think this approach is better, using the OSError and checking for the file exist errno for python2 support

File: PyInstaller/building/toc_conversion.py [hunk 2]
```diff
@@ -132,7 +132,11 @@ def __collect_data_files_from_zip(zipfilename):
         # 'PyInstaller.config' cannot be imported as other top-level modules.
         from ..config import CONF
         workpath = os.path.join(CONF['workpath'], os.path.basename(zipfilename))
-        os.makedirs(workpath)
+        try:
+            os.makedirs(workpath)
+        except OSError as e:
+            if e.errno != 17:
```
Comment (htgoebel) [inline PyInstaller/building/toc_conversion.py]: *shiver* Please use symbolic names, not dubious constant values. This is what the [manual](https://docs.python.org/2/library/exceptions.html#exceptions.OSError) recomments, too. `errno.EEXITS` ought to be the one you want to use here.

**Comment by drorasaf** [comment]
Fixed the constant issue, added import for errno only within the exception scope

**Comment by htgoebel** [comment]
Thanks. I'm looking forward to your next contribution :-)

**Comment by drorasaf** [comment]
It seems that it does not update it, don't know why
https://github.com/drorasaf/pyinstaller/commit/eb56ebd755de770837f08e61972fd107910faa17

**Pull request merged**